bridge-sdk/connectors/near-connector/src/near_connector.rs

```diff
@@ -0,0 +1,480 @@
+use bridge_connector_common::result::{BridgeSdkError, Result};
+use derive_builder::Builder;
+use near_contract_standards::storage_management::StorageBalance;
+use near_crypto::SecretKey;
+use near_primitives::{
+    hash::CryptoHash,
+    types::AccountId,
+    views::{FinalExecutionOutcomeView, FinalExecutionStatus},
+};
+use near_token::NearToken;
+use omni_types::{
+    locker_args::{BindTokenArgs, ClaimFeeArgs, DeployTokenArgs, FinTransferArgs},
+    ChainKind, Fee, OmniAddress,
+};
+use serde_json::json;
+use std::str::FromStr;
+
+/// Bridging NEAR-originated NEP-141 tokens
+#[derive(Builder, Default, Clone)]
+pub struct NearConnector {
+    #[doc = r"NEAR RPC endpoint"]
+    endpoint: Option<String>,
+    #[doc = r"NEAR private key"]
+    private_key: Option<String>,
+    #[doc = r"NEAR account id of the transaction signer"]
+    signer: Option<String>,
+    #[doc = r"Token locker account id on Near"]
+    token_locker_id: Option<String>,
+}
+
+impl NearConnector {
+    /// Logs token metadata to token_locker contract. The proof from this transaction is then used to deploy a corresponding token on Ethereum
+    #[tracing::instrument(skip_all, name = "LOG METADATA")]
+    pub async fn log_token_metadata(&self, near_token_id: String) -> Result<CryptoHash> {
+        let near_endpoint = self.endpoint()?;
+
+        let args = format!(r#"{{"token_id":"{near_token_id}"}}"#).into_bytes();
+
+        let tx_hash = near_rpc_client::change(
+            near_endpoint,
+            self.signer()?,
+            self.token_locker_id_as_str()?.to_string(),
+            "log_metadata".to_string(),
+            args,
+            300_000_000_000_000,
+            200_000_000_000_000_000_000_000,
+        )
+        .await?;
+
+        tracing::info!(tx_hash = tx_hash.to_string(), "Sent log transaction");
+
+        Ok(tx_hash)
+    }
+
+    /// Performs a storage deposit on behalf of the token_locker so that the tokens can be transferred to the locker. To be called once for each NEP-141
+    #[tracing::instrument(skip_all, name = "STORAGE DEPOSIT")]
+    pub async fn storage_deposit_for_token(
+        &self,
+        near_token_id: String,
+        amount: u128,
+    ) -> Result<CryptoHash> {
+        let near_endpoint = self.endpoint()?;
+        let token_locker = self.token_locker_id_as_str()?;
+
+        let args = format!(r#"{{"account_id":"{token_locker}"}}"#).into_bytes();
+
+        let tx_hash = near_rpc_client::change(
+            near_endpoint,
+            self.signer()?,
+            near_token_id,
+            "storage_deposit".to_string(),
+            args,
+            300_000_000_000_000,
```

Comment:
Const

Reply:
Fixed in https://github.com/Near-One/bridge-sdk-rs/pull/48/commits/2585f2e681cd2f0513f663227ff2ca10c6eefd5f